### PR TITLE
Add preferedScheme option

### PR DESCRIPTION
**Pull request by thejibz**
Issue #96

### CODE REVIEW

File: packages/oasgraph/src/oas_3_tools.ts [hunk 1]
```diff
@@ -194,36 +195,56 @@ export function getBaseUrl (
...
 
   throw new Error('Cannot find a server to call')
 }
 
+/**
+ * Return the first server that use the prefered scheme. If none match, return the first server.
+ * @param servers List of servers
+ * @param preferedScheme Optional. Prefered scheme to use
+ */
+function getPreferedServer(servers: ServerObject[], preferedScheme?: Schemes) {
```
Comment (Alan-Cha) [inline packages/oasgraph/src/oas_3_tools.ts]: I think this code can be expressed in a more intuitive way. What do you think about the following? 

```
function getPreferredServer(servers: ServerObject[], preferredScheme?: Schemes) {
  if (Array.isArray(servers)) {
    if (servers.length === 0) {
      // No choice in servers can be made
      return servers[0]
    } else if (servers.length >= 1) {
      let preferredServers = servers.filter(server => server.url && server.url.startsWith(`${preferredScheme}:`))
        
      if (preferredServers.length > 0) {
        if (preferredServers.length > 1) {
          logHttp(`Warning: Randomly selected the first server that match the preferredScheme option (${preferredScheme}): ${preferredServer.url}`)
        }

        return preferredServers[0]
      } else {
        // No preferred scheme could be found
        logHttp(`Warning: Randomly selected first server ${preferredServer.url}`)
        return servers[0]
      }
    }
  }
  throw new Error('Cannot find a server to call')
}
```

Comment (ErikWittern) [reply packages/oasgraph/src/oas_3_tools.ts]: In the above code, `if (servers.length === 0) {` (line 3) is wrong. The if statement tries to cover the case when there is only a single server, in which case `server.length` should be `1`, though, not `0`.

**Comment by thejibz** [comment]
> A few more comments... we appreciate your efforts to help make our style more consistent.
> 
> I think adding spaces within curly brackets, removing spaces before colons, and tabulating string concatentations is good. However, would it be possible to retain the space before parameters in function signature definitions?
> 
> For example: we would prefer `function sanitize (str: string): string` over `function sanitize(str: string): string`.
> 
> Also we’ve always used curly brackets for `if` and `else` statements.
> 
> Again, the fixes are welcomed but I think this is one of our better followed conventions.
> 
> In addition, would you be able to write some test cases? This is a bit tricky because this change cannot be tested in the same fashion as our usual Example API tests. If you’re looking for suggestions, I’m not sure what would be the best way to test for this but I think @erik may have some ideas. Perhaps we just need to test the function by itself rather than in the context of a working GraphQL interface.
> 
> Again, thanks a lot! Sorry for nitpicking!

No problem,  it's on me to adopt your code style ! I will change my formatter's settings in the context of your project. Also, I agree on the rule "always use curly brackets in if statements".

Regarding, the tests I didn't make one because of the reason you evoke. Not easy to do, but a unit testing on just the function will do the trick I think.

**Comment by thejibz** [comment]
I had to export the getPreferredServer function in order to test it though. Testing the n+1 getBaseUrl would be much harder.

**Comment by thejibz** [comment]
Hello guys !
I had to publish a npm package with the name "oasgraph-jibz" since the "git+https://..." style dependency don't work with the repo pattern root/package/oasgraph :/

I will delete this package as soon as you publish a new version with either preferredScheme or baseUrl option.

**Pull request merged**